jupyter/d2l-java/chapter_mlp/mlp.ipynb

```diff
@@ -0,0 +1,1006 @@
+    "We can easily design hidden nodes \n",
+    "to perform arbitrary computation,\n",
+    "for instance, basic logic operations on a pair of inputs.\n",
+    "Moreover, for certain choices of the activation function,\n",
+    "it is widely known that MLPs are universal approximators.\n",
+    "Even with a single-hidden-layer network,\n",
+    "given enough nodes (possibly absurdly many), \n",
+    "and the right set of weights,\n",
+    "we can model any function.\n",
+    "*Actually learning that function is the hard part.*\n",
+    "You might think of your neural network \n",
+    "as being a bit like the C programming language. \n",
+    "The language, like any other modern language,\n",
+    "is capable of expressing any computable program.\n",
+    "But actually coming up with a program \n",
+    "that meets your specifications is the hard part.\n",
+    "\n",
+    "Moreover, just because a single-layer network \n",
+    "*can* learn any function \n",
+    "does not mean that you should try \n",
+    "to solve all of your problems \n",
+    "with single-layer networks.\n",
+    "In fact, we can approximate many functions\n",
+    "much more compactly by using deeper (vs wider) networks.\n",
+    "We will touch upon more rigorous arguments in subsequent chapters,\n",
+    "but first let us actually build an MLP in code.\n",
+    "In this example, we’ll implement an MLP\n",
+    "with two hidden layers and one output layer.\n",
+    "\n",
+    "### Vectorization and Minibatch\n",
+    "\n",
+    "As before, by the matrix $\\mathbf{X}$, \n",
+    "we denote a minibatch of inputs.\n",
+    "The calculations to produce outputs \n",
+    "from an MLP with two hidden layers\n",
+    "can thus be expressed:\n",
+    "\n",
+    "$$\n",
+    "\\begin{aligned}\n",
+    "    \\mathbf{H}_1 & = \\sigma(\\mathbf{W}_1 \\mathbf{X} + \\mathbf{b}_1), \\\\\n",
+    "    \\mathbf{H}_2 & = \\sigma(\\mathbf{W}_2 \\mathbf{H}_1 + \\mathbf{b}_2), \\\\\n",
+    "    \\mathbf{O} & = \\mathrm{softmax}(\\mathbf{W}_3 \\mathbf{H}_2 + \\mathbf{b}_3).\n",
+    "\\end{aligned}\n",
+    "$$\n",
+    "\n",
+    "With some abuse of notation, we define the nonlinearity \n",
+    "$\\sigma$ to apply to its inputs in a row-wise fashion, \n",
+    "i.e., one observation at a time.\n",
+    "Note that we are also using the notation for *softmax* \n",
+    "in the same way to denote a row-wise operation.\n",
+    "Often, as in this section, the activation functions\n",
+    "that we apply to hidden layers are not merely row-wise, \n",
+    "but component wise.\n",
+    "That means that after computing the linear portion of the layer,\n",
+    "we can calculate each nodes activation \n",
+    "without looking at the values taken by the other hidden units.\n",
+    "This is true for most activation functions\n",
+    "(the batch normalization operation to be introduced in :numref:`sec_batch_norm` is a notable exception to that rule)."
```

Comment:
Can you add some information to introduce what you are trying to achieve?

Like: 'Here is some setup to run the code in the following sections"